ballista/scheduler/src/state/task_manager.rs

```diff
@@ -77,21 +77,20 @@ type ActiveJobCache = Arc<DashMap<String, JobInfoCache>>;
 
 #[derive(Default)]
 struct ActiveJobQueue {
-    queue: SegQueue<String>,
+    queue: Mutex<BinaryHeap<ActiveJob>>,
     jobs: ActiveJobCache,
 }
 
 impl ActiveJobQueue {
-    pub fn pop(&self) -> Option<ActiveJobRef> {
+    pub fn pop(&self) -> Option<JobInfoCache> {
         loop {
-            if let Some(job_id) = self.queue.pop() {
-                if let Some(job_info) = self.jobs.get(&job_id) {
-                    return Some(ActiveJobRef {
-                        queue: &self.queue,
-                        job: job_info.clone(),
-                        job_id,
-                    });
+            let mut queue_guard = self.queue.lock();
+            if let Some(active_job) = queue_guard.peek() {
```

Comment:
I think here you actually need to pop the job from the heap so you can update it after scheduling tasks